hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -210,15 +214,33 @@ private HoodieLogBlock readBlock() throws IOException {
   private HoodieLogBlock createCorruptBlock() throws IOException {
     log.info("Log " + logFile + " has a corrupted block at " + inputStream.getPos());
     long currentPos = inputStream.getPos();
-    long nextBlockOffset = scanForNextAvailableBlockOffset();
+    byte[] corruptedBytes;
+    int corruptedBlockSize = 0;
+    boolean done = true;
+    BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
```

Comment:
We can wrap this whole block in a CustomBufferedReader sort of implementation and just return the corruptedBytesSize on termination for cleanliness...

Reply:
Not sure if I fully follow.. But it would be good to have a Buffered, seekable abstraction over the log file.. Corrupt block handling etc should be left at this level IMO

Reply:
lets file a task for this?